ui/src/pages/version/components/forms/validation/schema.js

```diff
@@ -67,6 +67,21 @@ const environmentVariableSchema = yup.object().shape({
   value: yup.string(),
 });
 
+const stringValueSchema = (name) => yup.mixed().test(
+  `value input schema`,
+  `${name} must be a \`string\` but the final value was $\{value}`,
+  (value, { createError }) => {
+    try {
+      if (typeof value == "object" && "jsonPath" in value) {
+        return yup.string().required(`${name} is required`).validateSync(value["jsonPath"])
+      }
+      return yup.string().required(`${name} is required`).validateSync(value)
+    } catch (e) {
+      return createError({ message: e.message })
```

Comment:
Where is this `createError` function defined? :O

Reply:
It's from [yup.test() second argument ](https://www.npmjs.com/package/yup#schematestname-string-message-string--function--any-test-function-schema), and found out how to use this from [here](https://github.com/jquense/yup/issues/222#issuecomment-557457026) (but not using it with `this.` because I'm using arrow function instead of regular function) 

Backstory: docs said that the return value should be true or false or `ValidationError`. I kinda expect when I'm throwing a `ValidationError`, the error shown up in the UI would be this custom `ValidationError` I defined, but it shows nothing somehow 😅 turns out I need to use this `createError` for it to work

Reply:
Ahh got it, TIL hahaha 😅 Thanks for sharing that. Just one more short question though, how is the error message that shows up different if you were to `return false` vs `return createError({ message: e.message })`? 🤔

Reply:
So, when this func return `false` (tho none here will return false) it's going to show the default `"${name} must be a string but the final value was $\{value}"` error I put as the message argument. If I throw e.g. `createError({ message: "new error"})`, it will show `"new error"` as the message instead

Now, I've mentioned that the function will actually never return a `false`, it's because `.validateSync()` will either return the value or `ValidationError`. When `.validateSync()` throw an error, I'll simply throw the error again (correctly using the `createError`), so the value input check will has the same error message for every field in [here](https://github.com/caraml-dev/merlin/blob/da2803ab384a83d352663100f299f2c0699836dd/ui/src/pages/version/components/forms/validation/schema.js#L113-L117)

example of before using `createError` (I return false if there's an error) vs using `createError` 

<img width="618" alt="Screenshot 2024-08-29 at 14 42 29" src="https://github.com/user-attachments/assets/e6645e29-750c-4f0d-94c3-4d9dffbae978">

<img width="561" alt="Screenshot 2024-09-02 at 14 23 33" src="https://github.com/user-attachments/assets/07390f68-a687-4044-b045-31b65dc92feb">

Reply:
Ahh got it, thanks for the really concise and clear explanation! I was about to ask if using a simpler error message would've made it simpler 😅 like use `String input value is required` instead of `${name} must be a string but the final value was $\{value}` since it seems like the `value` variable can't be parsed correctly (weird behaviour I don't know why this happens with `yup`) when a `ValidationError` is thrown. But your implementation works too, thanks a lot for investigating this!